app/models/edition.rb

```diff
@@ -179,4 +180,8 @@ def assign_revision(revision, user)
 
     self
   end
+
+  def editors
+    (revisions + statuses).map(&:created_by).uniq
```

Comment:
This would cause one query for every revision and status so I think we should refactor it as that could be a ton of queries.

I did some experimenting to see if it could go into a `has_many` but I hit issues with that. This was as far as I got:

```
  has_many :editors,
           -> (edition) {
            user_ids = edition.statuses.pluck(:created_by_id) + edition.revisions.pluck(:created_by_id)
            where(id: user_ids).distinct
           },
           class_name: "User"
```

But I couldn't kill a scope of where `edition_id = edition.id` so that wasn't going anywhere. I think the right way for a has_many would be denormalising it anyway.

So could use that technique in this method to reduce the queries. 

I do feel like we might possibly hit a reason that we do want to denormalise this at some point though. As it feels a bit delicate that this pulls data off immutable data and thus couldn't be altered.

Reply:
The scope looks pretty good to me, as it bounds the number of queries. However, I'm assuming it will be advantageous to denormalise the editors in order to filter by them on the index page. Would a workable compromise be to use the scope for now, but maybe do a separate PR where we look at making editors a persisted concept?

Reply:
Yup that sounds good, so we'll plan to denormalise as part of index?

That scope I wrote doesn't work, it'd have to be in your method by the way.

Reply:
Ah, I see what you mean now - that's probably for the best :-). I've added a note to the docs section of the card about searching for editors.